src/dlgProfilePreferences.cpp

```diff
@@ -2085,6 +2111,79 @@ void dlgProfilePreferences::slot_logFileNameFormatChange(const int index)
     label_logFileNameExtension->setVisible(isShown);
 }
 
+void dlgProfilePreferences::slot_configDirOptionChange(const int index)
+{
+    Q_UNUSED(index);
+
+    // Show options to set the config directory if 'Select a
+    // directory...' is selected
+    bool isShown = index == 1;
+    label_setConfigDir->setVisible(isShown);
+    lineEdit_configDir->setVisible(isShown);
+    toolButton_setConfigDir->setVisible(isShown);
+
+    switch (index) {
+    case 0:
+        // Clear mConfigDirPath if 'Default' is selected
+        mConfigDirPath.clear();
+        lineEdit_configDir->setText(QDir::homePath() + "/.config/mudlet");
+        break;
+    case 1:
+        // Set mConfigDirPath to the value of lineEdit_configDir if
+        // 'Select a directory' is selected
+        mConfigDirPath = lineEdit_configDir->text();
+        break;
+    case 2:
+        // Set mConfigDirPath to a relative path if 'Relative to the
+        // application file' is selected
+        mConfigDirPath = ".";
+        lineEdit_configDir->setText(QCoreApplication::applicationDirPath() + "/config");
+        break;
+    }
+}
+
+void dlgProfilePreferences::slot_setConfigDir()
+{
+    QString startDir;
+    if (lineEdit_configDir->text().isEmpty() ||
+        lineEdit_configDir->text() == QString("%1/%2").arg(QDir::homePath(), ".config/mudlet")) {
+        // Set the directory the QFileDialog starts in to the default
+        // directory if the line edit is empty or equal to Mudlet's
+        // default configuration directory
+        startDir = QDir::homePath() + "/.config";
+    } else if (lineEdit_configDir->text() ==
+               QString("%1/%2").arg(QCoreApplication::applicationDirPath(), "config")) {
+        // Use the contents of the line edit if set to the path
+        // relative to the application file
+        startDir = lineEdit_configDir->text();
+    } else {
+        // Start in the parent directory if a custom directory is set
+        QFileInfo configDirInfo(lineEdit_configDir->text());
+        QDir configDirParent(configDirInfo.dir());
+        startDir = configDirParent.absolutePath();
+    }
+
+    QString dir = QFileDialog::getExistingDirectory(this,
+                                                    tr("Select a directory where your configuration will be stored"),
+                                                    startDir,
+                                                    QFileDialog::DontUseNativeDialog);
```

Comment:
:+1: for `QFileDialog::DontUseNativeDialog` as on Windows, the Native one hides the display of the ***files*** in the directory!

Reply:
Why should we show files when the user is picking a directory and not a file?

Reply:
But if all dialogs on Windows when picking folders do this - why would they think they selected an empty directory?

Reply:
It's like we're arguing with the UX designers of Windows as to what is better for their users. I think it's better to leave it up to the experts in their domain here...